admin/updates/20190619-similarity.sql

```diff
@@ -0,0 +1,40 @@
+BEGIN;
+
+-- CREATE EXTENSION IF NOT EXISTS "cube"; -- Only required for postgres similarity
+
+CREATE TABLE similarity (
+  id INTEGER -- PK, FK to lowlevel
+);
+
+ALTER TABLE similarity ADD CONSTRAINT similarity_pkey PRIMARY KEY (id);
+
+ALTER TABLE similarity
+  ADD CONSTRAINT similarity_fk_lowlevel
+  FOREIGN KEY (id)
+  REFERENCES lowlevel (id);
+
+CREATE TABLE similarity_metrics (
+  metric TEXT, -- PK
```

Comment:
I’d like to know the size of these tables (because this is going to have an impact on the full database size). Can you run some disk usage statistics on your subset for the lowlevel table and these tables when they’re full? That way we should be able to say something like “the similarity tables will be x% of the lowlevel tables”

Reply:
This is a pastebin with all of the general table size information, should I extract anything more specific to the similarity and lowlevel tables, or is this what you had in mind?
https://pastebin.com/Wxg7fBKV

Reply:
I think this https://pastebin.com/KxLFKZMS also gives a more simple and clear depiction.

Reply:
With that in mind, the similarity table is about 9 times as large as the lowlevel_json table.

Reply:
The json data in the lowlevel_json table is actually stored in `pg_toast.pg_toast_196767`. This is a side effect of the way that postgres stores the data (the ‘total’ column of the first paste). So we’re actually looking at about 30x smaller for the similarity table. This means that this table will be able 15gb in size with 450gb of items.

Reply:
Aha... I had a feel that didn’t really make much sense😅. So does this impact our full database size in a way that requires us to make any changes?